### DISCUSSION TITLE
Linting only for changed files

### DISCUSSION THREAD

**Discussion by anckaert1998** [nrwl/nx#19900] [Feature Requests]
Dear

When we currently perform linting (via affected:lint) this lints all the files in all the affected libraries, but I don't see the point in this.
When a file has not been changed there is no use in lint that file because it can not fail.
So I am wondering what the reasoning behind it is or if I'm missing some information

**Comment by ahernd2** [comment]
We have a similar scenario. We are using sonar analysis on each microfrontend. This is particularly annoying because when a developer changes a file that affects lots of projects, sonar analysis runs for each affected one.

Sonar analysis is very slow and cannot be parallelized like lint can. Additionally it is wasteful as running the analysis against the affected projects provides no advantage. Only the project where the file was changed is the only project that should be scanned.

It would be good to provide a way to tell nx I only want the affected projects that have a physical file change for a particular task.

**Comment by christopher-buss** [comment]
For me, I have a local test runner that uses .ts config files to set it up for our other packages. Despite the fact that this project changing can only cause another packages .ts config file to potentially go out of date lint wise, I have to pay the cost of linting the entire package because it gets marked as affected.
